ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -320,12 +405,17 @@ private void addAtlasDumpTask(boolean bootstrap, Path prevHiveDumpDir) {
 
 
   private void finishRemainingTasks() throws SemanticException {
-    Path dumpAckFile = new Path(work.getCurrentDumpPath(),
-            ReplUtils.REPL_HIVE_BASE_DIR + File.separator
-                    + ReplAck.DUMP_ACKNOWLEDGEMENT.toString());
+    boolean isFailoverInProgress = conf.getBoolVar(HiveConf.ConfVars.HIVE_REPL_FAILOVER_START) && !work.isBootstrap();
+    if (isFailoverInProgress) {
+      Utils.create(new Path(work.getCurrentDumpPath(), ReplUtils.REPL_HIVE_BASE_DIR + File.separator
+              + ReplAck.FAILOVER_READY_MARKER), conf);
+      LOG.info("Dump marked as failover ready");
+    }
+    Path dumpAckFile = new Path(work.getCurrentDumpPath(), ReplUtils.REPL_HIVE_BASE_DIR + File.separator
+                    + ReplAck.DUMP_ACKNOWLEDGEMENT);
     Utils.create(dumpAckFile, conf);
     prepareReturnValues(work.getResultValues());
-    work.getMetricCollector().reportEnd(Status.SUCCESS);
+    work.getMetricCollector().reportEnd(isFailoverInProgress ? Status.FAILOVER_READY : Status.SUCCESS);
```

Comment:
Status.FAILOVER_READY sounds like Status.SUCCESS + failover ready state. I think it should have been additional state field for FAILOVER_STATE:STARTED/COMPLETED/ABORTED? kind of